lib/pf/Authentication/Source/EAPTLSSource.pm

```diff
@@ -0,0 +1,104 @@
+package pf::Authentication::Source::EAPTLSSource;
+
+=head1 NAME
+
+pf::Authentication::Source::EAPTLSSource add documentation
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::Authentication::Source::EAPTLSSource
+
+=cut
+
+use strict;
+use warnings;
+use Moose;
+use pf::constants;
+use pf::Authentication::constants;
+use List::Util qw(first);
+
+extends 'pf::Authentication::Source';
+
+has '+class' => (default => 'external');
+has '+type' => (default => 'TLS');
+
+=head2 available_attributes
+
+Allow to make a condition on the user's email address.
+
+=cut
+
+sub available_attributes {
+    my ($self) = @_;
+    my $super_attributes = $self->SUPER::available_attributes;
+    my @own_attributes = map { {value => $_, type => $Conditions::SUBSTRING}} qw(
+      TLS-Client-Cert-Serial
```

Comment:
Would it make more sense to revoke the cert in question and refuse it? Or refuse certs that were delivered before X date? From what I found, depending on the configuration of the CA, it can choose the serial randomly.